rust_src/src/marker.rs

```diff
@@ -102,4 +107,63 @@ pub fn set_point_from_marker(marker: LispMarkerRef) {
     unsafe { set_point_both(charpos, bytepos) };
 }
 
+/// Return insertion type of MARKER: t if it stays after inserted text.
+/// The value nil means the marker stays before text inserted there.
+#[lisp_fn]
+pub fn marker_insertion_type(marker: LispMarkerRef) -> bool {
+    marker.insertion_type()
+}
+
+/// Set the insertion-type of MARKER to TYPE.
+/// If ITYPE is non-nil, it means the marker advances when you insert text at it.
+/// If ITYPE is nil, it means the marker stays behind when you insert text at it.
+#[lisp_fn]
+pub fn set_marker_insertion_type(mut marker: LispMarkerRef, itype: LispObject) -> LispObject {
+    unsafe { mset_insertion_type(marker.as_mut(), itype.is_not_nil() as BoolBF) };
+    itype
+}
+
```

Comment:
`itype` is actually just a boolean then and this function returns a boolean.

Reply:
No, it can be nil and non-nil. I also changed the docstring as it was wrong.

Reply:
Looking at the lisp code, t and nil are the usual parameters.

Reply:
Right. In Lisp NIL is false and anything else is true.